requirements.txt

```diff
@@ -0,0 +1,2 @@
+dockerspawner
+jupyterhub-idle-culler==1.2.1
```

Comment:
In fact, I prefer to version pinning here.

Otherwise, we will need a separate commit to manually bump each version to latest before releasing a new JupyterHub image.  To much work, unless we actually need this level of reproducibility.

Currently our lighter reproducibility option is to use an older image if need be.

Reply:
What happens when one of the dependencies makes a breaking change in the latest version breaks this image though?
Wouldn't you rather be in control of that?

Reply:
I assume this breaking change happens rarely so I would rather have 10 smooth sail build to 1 breaking build and just wait out a bit for the compat issue to be fixed and trigger another rebuild.

You are right that if breaking changes happen very frequently when we need to pin down every dependencies to be sure.

Speaking of pinning down dependencies, we should also pin down all indirect dependencies ala `pip freeze` to really be sure for reproducible builds.

Reply:
I agree with @tlvu. The Docker images are pinned, so unless we purposely override the tags, reproductibility is obtained there. I prefer to gradually receive fixes/improvements/security resolutions silently over manually managing all of these dependencies across all repos.